internal/service/resourceexplorer2/search_data_source.go

```diff
@@ -0,0 +1,184 @@
+					Attributes: map[string]schema.Attribute{
+						"complete": schema.BoolAttribute{
+							Computed: true,
+						},
+						"total_resources": schema.Int64Attribute{
+							Computed: true,
+						},
+					},
+				},
+			},
+			"resources": schema.ListNestedBlock{
+				CustomType: fwtypes.NewListNestedObjectTypeOf[resourcesData](ctx),
+				NestedObject: schema.NestedBlockObject{
+					Attributes: map[string]schema.Attribute{
+						"arn": schema.StringAttribute{
+							Computed: true,
+						},
+						"last_reported_at": schema.StringAttribute{
+							Computed: true,
+						},
+						"owning_account_id": schema.StringAttribute{
+							Computed: true,
+						},
+						"region": schema.StringAttribute{
+							Computed: true,
+						},
+						"resource_type": schema.StringAttribute{
+							Computed: true,
+						},
+						"service": schema.StringAttribute{
+							Computed: true,
+						},
+					},
+					Blocks: map[string]schema.Block{
+						"resource_property": schema.ListNestedBlock{
+							CustomType: fwtypes.NewListNestedObjectTypeOf[resourcePropertyData](ctx),
+							NestedObject: schema.NestedBlockObject{
+								Attributes: map[string]schema.Attribute{
+									"data": schema.StringAttribute{
+										Computed: true,
+										// Validators: jsonvalidator.JSON(),
+									},
+									"last_reported_at": schema.StringAttribute{
+										Computed: true,
+									},
+									"name": schema.StringAttribute{
+										Computed: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (d *dataSourceSearch) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	conn := d.Meta().ResourceExplorer2Client(ctx)
+
+	var data dataSourceSearchData
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	data.ID = types.StringValue(data.ViewArn.ValueString())
+
+	input := &resourceexplorer2.SearchInput{
+		QueryString: aws.String(data.QueryString.ValueString()),
+	}
+
+	paginator := resourceexplorer2.NewSearchPaginator(conn, input)
+
+	var out resourceexplorer2.SearchOutput
+	for paginator.HasMorePages() {
+		page, err := paginator.NextPage(ctx)
+		if err != nil {
+			resp.Diagnostics.AddError(
+				create.ProblemStandardMessage(names.ResourceExplorer2, create.ErrActionReading, DSNameSearch, data.ID.String(), err),
+				err.Error(),
+			)
+			return
+		}
+
+		if page != nil && len(page.Resources) > 0 {
+			out.Resources = append(out.Resources, page.Resources...)
+		}
+	}
+
+	resp.Diagnostics.Append(flex.Flatten(ctx, out, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+type dataSourceSearchData struct {
+	ResourceCount fwtypes.ListNestedObjectValueOf[countData]     `tfsdk:"resource_count"`
+	Resources     fwtypes.ListNestedObjectValueOf[resourcesData] `tfsdk:"resources"`
+	MaxResults    types.Int64                                    `tfsdk:"max_results"`
```

Comment:
Since we don't include this attribute in the Terraform schema, this can be deleted. It won't affect flattening of the AWS structure.

```suggestion
```